From 35b1ae6cdb8af8c3b3283e84c0f55ca03f1a2e85 Mon Sep 17 00:00:00 2001
From: "mrobinson@webkit.org"
 <mrobinson@webkit.org@268f45cc-cd09-0410-ab3c-d52691b4dbfc>
Date: Mon, 26 Nov 2012 22:00:15 +0000
Subject: [PATCH] [GTK] Explicitly link against librt
 https://bugs.webkit.org/show_bug.cgi?id=103194

Patch by Kalev Lember <kalevlember@gmail.com> on 2012-11-26
Reviewed by Martin Robinson.

Fixes broken build with undefined references to shm_open / shm_unlink
symbols. SharedMemoryUnix.cpp uses these so we need to link with -lrt.

.:

* configure.ac:

Source/WebKit2:

* GNUmakefile.am:

git-svn-id: http://svn.webkit.org/repository/webkit/trunk@135761 268f45cc-cd09-0410-ab3c-d52691b4dbfc
---
 ChangeLog                     | 12 ++++++++++++
 Source/WebKit2/ChangeLog      | 12 ++++++++++++
 Source/WebKit2/GNUmakefile.am |  1 +
 configure.ac                  |  7 +++++++
 4 files changed, 32 insertions(+)

diff --git a/Source/WebKit2/GNUmakefile.am b/Source/WebKit2/GNUmakefile.am
index 097cdc1..ff5ff1f 100644
--- a/Source/WebKit2/GNUmakefile.am
+++ b/Source/WebKit2/GNUmakefile.am
@@ -566,6 +566,7 @@ Programs_WebKitPluginProcess_LDADD += \
 	$(PANGO_LIBS) \
 	$(PNG_LIBS) \
 	$(SHLWAPI_LIBS) \
+	$(SHM_LIBS) \
 	$(SQLITE3_LIBS) \
 	$(UNICODE_LIBS) \
 	$(XRENDER_LIBS) \
diff --git a/configure.ac b/configure.ac
index 4e1f0e8..97980e4 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1118,6 +1118,13 @@ if test "$enable_webkit2" = "yes"; then
    if test "$have_gtk_unix_printing" = "yes"; then
        AC_DEFINE([HAVE_GTK_UNIX_PRINTING], [1], [Define if GTK+ UNIX Printing is available])
    fi
+
+   # On some Linux/Unix platforms, shm_* may only be available if linking
+   # against librt
+   if test "$os_win32" = "no"; then
+       AC_SEARCH_LIBS([shm_open], [rt], [SHM_LIBS="-lrt"])
+       AC_SUBST(SHM_LIBS)
+   fi
 fi
 
 # Plugin Process
-- 
1.8.0.1

